### PR TITLE
[KYUUBI #3531] [DOC] Add documentation for Hive connector

### PR DESCRIPTION
**Pull request by Yikf**
### _Why are the changes needed?_

Fix https://github.com/apache/incubator-kyuubi/issues/3531

This pr aims to add documentation for Hive connector

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

File: docs/connector/spark/hive.rst
Comment (yaooqinn) [inline docs/connector/spark/hive.rst:1]: the copyright you have copied from other rst is malformed by you editor, please send a follow up to fix ,FYI
https://kyuubi.readthedocs.io/en/latest/connector/spark/hive.html

Comment (yaooqinn) [reply docs/connector/spark/hive.rst:1]: Hi @Yikf. It's recommended to add screenshots for manual local doc build if appropriate, and it's better for a committer to ask for it before merging, cc @pan3793

**Pull request closed without merging**
